### ISSUE TITLE
Support DRM

### ISSUE THREAD

**Issue by bqv** [atlas-engineer/nyxt#1041]
**Describe the bug**
Noting this more for logging purposes, more than requesting a fix.
(And frankly, I wouldn't blame you if you made this a wontfix)

**Precise recipe to reproduce the issue**
Attempt to use e.g. Amazon Prime Video -> error due to unsupported form of DRM

**Information**
- OS name+version: Nixos@2020-11-05
- Graphics card and driver: amdgpu
- Desktop environment / Window manager name+version: long story
- How you installed Nyxt (Guix pack, package manager, build from source): nix
- Information from command copy-system-information: Nyxt version: 2 
Operating system kernel: Linux 5.8.14
Lisp implementation: SBCL 2.0.8.nixos
Features: (WEBKIT2 WEBKIT2-2.28 WEBKIT2-EMOJI WEBKIT2-MEDIA WEBKIT2-SANDBOXING
           GTK-3-22 GTK-3-20 GTK-3-18 GTK-3-16 GTK-3-14 GTK-3-12 GTK-3-10
           GTK-3-8 GTK-3-6 GTK-3-4 GTK GDK-3-22 GDK-3-20 GDK-3-18 GDK-3-16
           GDK-3-14 GDK-3-12 GDK-3-10 GDK-3-8 GDK-3-6 GDK-3-4 CAIRO-1-10
           CAIRO-1-12 GDK-PIXBUF GLIB-2-30 GLIB-2-32 GLIB-2-34 GLIB-2-36
           GLIB-2-38 GLIB-2-40 GLIB-2-42 GLIB-2-44 GLIB-2-46 GLIB-2-48
           GLIB-2-50 GLIB-2-52 GLIB-2-54 GLIB-2-56 GLIB-2-58 GLIB
           FSET-EXT-STRINGS SWANK PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES
           PARENSCRIPT NAMED-READTABLES OSICAT-FD-STREAMS 21BIT-CHARS CHUNGA
           FLEXI-STREAMS CLOSER-MOP CL-PPCRE-UNICODE CL-UNICODE CL-PPCRE
           COMPARE-AND-SWAP-VOPS BORDEAUX-THREADS SPLIT-SEQUENCE GRAY-STREAMS
           FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE SEQUENCE-EMPTYP
           FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE ASDF-SYSTEM-CONNECTIONS
           CL-JSON-CLOS CL-JSON THREAD-SUPPORT ASDF3.3 ASDF3.2 ASDF3.1 ASDF3
           ASDF2 ASDF OS-UNIX NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 GENCGC
           64-BIT ANSI-CL COMMON-LISP ELF IEEE-FLOATING-POINT LINUX
           LITTLE-ENDIAN PACKAGE-LOCAL-NICKNAMES SB-LDB SB-PACKAGE-LOCKS
           SB-THREAD SB-UNICODE SBCL UNIX)

**Comment by Ambrevar** [comment]
Ha, indeed the "wontfix" is hanging close... :D

But first, let's see if there is a simple fix at zero cost.

Can you reproduce the issue on Epiphany?
Can you also test with the latest Qutebrowser (should be running
QtWebengine)?

**Comment by bqv** [comment]
Can reproduce with Qute, I don't have epiphany but it's a fair assumption it would also do this.

FWIW from previous investigation I know it's a limitation of webkitgtk, but the functionality exists in a branch or feature flag somewhere (something about Encrypted Media Extensions), I just never figured out how to get this to use that

**Comment by aartaka** [comment]
@bqv, can you please test on `master` (or on the next release, whatever is fine by you)? We enabled WebKit permission requests in #1849, and that had something about EME, so it well bay be related to what you want :)

**Comment by bqv** [comment]
In time, shall do!

**Comment by aartaka** [comment]
@bqv, you might need to put this snippet in your config to enable support for EME:
``` lisp
(defmethod ffi-buffer-make :after ((buffer gtk-buffer))
  (let ((settings (webkit:webkit-web-view-get-settings (nyxt::gtk-object buffer))))
    (setf
     (webkit:webkit-settings-enable-encrypted-media settings) t)))
```

Note that WebKitGTK should be compiled with `ENABLE_ENCRYPTED_MEDIA` flag for this to work.

**Comment by aartaka** [comment]
@aadcg, why complete? Maybe we should enable it by default so that the issue is properly addressed?

**Comment by aartaka** [comment]
> Besides not being a priority, this is not the best time to introduce renderer-specific logic given that we're in the midst of supporting another renderer.

It's well-contained and is as easy to enable as the console message logging. In the same way and the same place, actually. Nothing hurting portability and bloating the foreign interface.

**Comment by jmercouris** [comment]
Right, but the renderers are in flux right now, this is something that we probably won't do and don't need to think about for at least a year or so.

**Comment by aartaka** [comment]
Wait, we already have encrypted media enabled on GTK port. Well, I guess we don't have to do anything regarding this issue then. Sorry for changing the status again, but it really is completed now 😅
